### PR TITLE
fix(metrics): Always include email in traits for identify calls

### PR DESCRIPTION
**Pull request by tianrunhe**
# Description

While investigating Segment <-> HubSpot integration, I saw this error pop out from Segment that complains `email` is missing from identify calls:

Screenshot 2022-12-05 at 10 03 46 AM

## Demo

Screenshot 2022-12-05 at 2 07 12 PM

## Testing scenarios
- [ ] Make sure `email` is included in `traits` for all `identify` calls

## Final checklist

- [x] I checked the [code review guidelines](../docs/codeReview.md)
- [x] I have added [Metrics Representative](../docs/codeReview.md#metrics-representative) as reviewer(s) if my PR invovles metrics/data/analytics related changes
- [x] I have performed a self-review of my code, the same way I'd do it for any other team member
- [x] I have tested all cases I listed in the testing scenarios and I haven't found any issues or regressions
- [x] Whenever I took a non-obvious choice I added a comment explaining why I did it this way
- [x] I added the label `One Review Required` if the PR introduces only minor changes, does not contain any architectural changes or does not introduce any new patterns and I think one review is sufficient'
- [x] PR title is human readable and could be used in changelog

### CODE REVIEW

**Comment by Dschoordsch** [comment]
Sorry I haven't looked into it yet. Will review it later today

File: packages/server/billing/helpers/adjustUserCount.ts [hunk 1]
```diff
@@ -150,8 +146,12 @@ export default async function adjustUserCount(
   const r = await getRethink()
   const orgIds = Array.isArray(orgInput) ? orgInput : [orgInput]
 
+  const user = await getUserById(userId)
+  if (!user) {
+    throw new Error(`User does not exist: ${userId}`)
+  }
   const dbAction = dbActionTypeLookup[type]
-  await dbAction(orgIds, userId)
+  await dbAction(orgIds, user)
```
Comment (Dschoordsch) [inline packages/server/billing/helpers/adjustUserCount.ts:154]: +1 I would prefer if we'd pass just `userId, email` to make clear that's the only thing we're using from the object. No big deal though.

File: packages/server/billing/helpers/adjustUserCount.ts [hunk 2]
```diff
@@ -150,8 +146,12 @@ export default async function adjustUserCount(
   const r = await getRethink()
   const orgIds = Array.isArray(orgInput) ? orgInput : [orgInput]
 
+  const user = await getUserById(userId)
```
Comment (Dschoordsch) [inline packages/server/billing/helpers/adjustUserCount.ts:149]: 11 I think we can safely use the dataloader here. While we might mutate the `OrganizationUser` we don't touch the `User` object here or in the mutations, so there is no risk of reading stale data.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
Something seems amiss with `adjustUserCount`. If I remove a member from my org, I see a segment event for their new identity but none for the remove event or `adjustUserCount`. Do you know what's happening there?

File: packages/server/billing/helpers/adjustUserCount.ts (hunk 1 above)
Comment (tianrunhe) [reply packages/server/billing/helpers/adjustUserCount.ts:154]: I have other refactor coming up that needs the whole user object. You can check it out later!

**Comment by tianrunhe** [comment]
> I see a segment event for their new identity but none for the remove event or adjustUserCount

The `removeFromOrg` triggers db action type of `REMOVE_USER` which goes to `deleteUser` and we didn't have any metrics emitted there. I added a new one.

**Review by Dschoordsch — CHANGES_REQUESTED** [review]
You missed adding the dataloader to `adjustUserCount` in 2 places
https://github.com/ParabolInc/parabol/blob/c7f68af4813448e4ec8ccbdcaa45a9e999260cff/packages/server/safeMutations/acceptTeamInvitation.ts#L96
https://github.com/ParabolInc/parabol/blob/c7f68af4813448e4ec8ccbdcaa45a9e999260cff/packages/server/graphql/private/mutations/autopauseUsers.ts#L29
No idea why this isn't caught by `yarn typecheck`

**Review by Dschoordsch — APPROVED** [review]

**Pull request merged**